### PR TITLE
Provide actionable error message when RESTEasy can't find JSON serializer

### LINKED ISSUE #19756: Replace "Could not find MessageBodyWriter for response object of type: java.util.ArrayList of media type: application/json" with helpful tip
### Description

if you forget to add quarkus-resteasy-jackson or quarkus-resteasy-jsonb and try open rest endpoint returning json you just get a single line:

`Could not find MessageBodyWriter for response object of type: java.util.ArrayList of media type: application/json`

Could we be helpful and give a hint to user that what is missing is a json handler, probably missing quarkus-resteasy-jackon or quarkus-resteasy-jsonb ....

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Resolves: #19756

### CODE REVIEW

File: extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NoMessageBodyWriterFoundFailureExceptionMapper.java
Comment (gsmet) [inline extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NoMessageBodyWriterFoundFailureExceptionMapper.java:23]: We already do something about it here: https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-classic/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java#L220 . And also for Mutiny types.

Not sure if it's worth adding more?

**Review by gsmet — CHANGES_REQUESTED** [review]
Asked a question. Let's figure it out before merging.

File: extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NoMessageBodyWriterFoundFailureExceptionMapper.java
Comment (geoand) [reply extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NoMessageBodyWriterFoundFailureExceptionMapper.java:23]: Hm... I wasn't aware of that. Yeah, it's probably a better approach. Let me check

Comment (geoand) [reply extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NoMessageBodyWriterFoundFailureExceptionMapper.java:23]: I actually do see `2021-08-30 12:55:06,338 WARN  [io.qua.res.com.dep.ResteasyCommonProcessor] (build-4) Quarkus detected the need of REST JSON support but you have not provided the necessary JSON extension for this. You can visit https://quarkus.io/guides/rest-json for more information on how to set one.
` in the logs now that you mention it.

I'll leave it up to you and @maxandersen to decide whether this provides additional benefit or not.

Comment (maxandersen) [reply extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NoMessageBodyWriterFoundFailureExceptionMapper.java:23]: The issue is that the error you get when using curl or browser to access the resource does not tell you what is the cause of error.

At least in devmode we should be able to show more informative error page.

Comment (geoand) [reply extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NoMessageBodyWriterFoundFailureExceptionMapper.java:23]: What I did is indeed dev-mode only

**Comment by gastaldi** [comment]
Will this also be needed for RestEasy Reative?

**Comment by geoand** [comment]
Yeah, we can do simithing similar, but the implementation will be completely different

**Comment by geoand** [comment]
@maxandersen do we want this in or not after all?

**Comment by geoand** [comment]
Should I close this or do we want it after all?

**Comment by geoand** [comment]
Closing this for now

**Comment by ahofmeister** [comment]
@geoand I think this would be useful. I have added `quarkus-jackson` and `quarkus-resteasy` before I added the right one: `quarkus-resteasy-jackson`. This is not a big deal, as I knew I had to add some other dependencies. But even after setting up a ton of Quarkus services, it got me confused - all the time. So for less experienced people with Quarkus, RESTEasy and Jackson, the actual error message might be not on spot.

**Pull request closed without merging**
